### PR TITLE
test: Test quality aggregation of total consumption

### PR DESCRIPTION
**Pull request by BjarkeMeier**
The PR also introduces generalized factories for energy result rows and data frames.

Sut:
image

Test organization:
image

PyCharm test run:
image

### CODE REVIEW

File: source/databricks/calculation_engine/tests/calculation/energy/energy_results.py
Comment (energinet-ajw) [inline source/databricks/calculation_engine/tests/calculation/energy/energy_results.py:1]: Would it be better to call this file something like energy_result_row_builder.py?

File: source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+)
+def test__when_valid_input__returns_expected_qualities(
+    spark: SparkSession,
+    prod_qualities: list[QuantityQuality],
+    exchange_qualities: list[QuantityQuality],
+    expected_qualities: list[str],
+) -> None:
+    """
+    Test that qualities from both production and net exchange is aggregated.
+    """
```
Comment (energinet-ajw) [inline source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py:50]: Good with a comment. Could it be part of the method name instead?

File: source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    production = energy_results.create_row(qualities=prod_qualities)
+    net_exchange = energy_results.create_row(qualities=exchange_qualities)
+    production_per_ga = energy_results.create(spark, production)
+    net_exchange_per_ga = energy_results.create(spark, net_exchange)
+
+    # Act
+    actual = sut(production_per_ga, net_exchange_per_ga)
+
+    # Assert
+    actual.df.show()
```
Comment (energinet-ajw) [inline source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py:61]: Don't think line 61 is needed.

File: source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py (hunk 1 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py]: That will cause pretty long names, but I tend to agree with you.
I've experimented with test classes. What do you think?

File: source/databricks/calculation_engine/tests/calculation/energy/energy_results.py
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/tests/calculation/energy/energy_results.py:1]: Probably. I went with energy_result_factories. Ok?

Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/calculation/energy/energy_results.py:1]: ok

File: source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py (hunk 1 above)
Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/calculation/energy/grid_loss_calculator/test_calculate_total_consumption.py]: As a new developer I think it's great. It gives context. If the rest of the team agree I say go for it.

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**